onedal/neighbors/neighbors.py

```diff
@@ -131,20 +131,20 @@ def _get_onedal_params(self, X, y=None):
             "method": self._fit_method,
             "radius": self.radius,
             "class_count": class_count,
-            "neighbor_count": self.n_neighbors,
+            "neighbor_count": self.n_neighbors if n_neighbors is None else n_neighbors,
```

Comment:
```suggestion
            "neighbor_count": int(self.n_neighbors if n_neighbors is None else n_neighbors),
```

Reply:
Handled by parameter_constraints in sklearn: https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/neighbors/_base.py#L396 Not only does it need to be an int, it must be >0